CMakeLists.txt

```diff
@@ -1,9 +1,14 @@
-# =============================================================================
-# Copyright (C) 2018-2022 Blue Brain Project
+# Copyright 2023 Blue Brain Project, EPFL.
 #
-# This file is part of NMODL distributed under the terms of the GNU Lesser General Public License.
-# See top-level LICENSE file for details.
-# =============================================================================
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
```

Comment:
@alkino : I haven't looked at the details myself - do we need to keep/maintain license text in each source file? I was quickly reading this https://softwareengineering.stackexchange.com/questions/317041/should-i-add-the-license-in-every-header-and-source-file.

If this is not required, should we just get rid of it?

Reply:
More suggestions in https://opensource.stackexchange.com/questions/202/should-i-include-license-text-in-a-single-file-or-all-source-files

Reply:
I can do what you decide but: https://www.apache.org/licenses/LICENSE-2.0#apply

Reply:
@alkino - having this whole boilerplate everywhere is bit too much noisy? What if we just do what we did for [coreneuron](https://github.com/neuronsimulator/nrn/blob/master/src/coreneuron/mpi/nrnmpi.h) or what is done in [spack](https://github.com/BlueBrain/spack/blob/develop/bin/spack-python)? i.e. 

```console

/*
 * Copyright 2023 Blue Brain Project, EPFL.
 *
 * See the top-level LICENSE file for details.
 */
````
what do you think?

Reply:
I agree

Reply:
There have been several discussions about this, but I'm not sure what the final allowances were.  I *think* convinced people that the `SPDX` style were ok, but it's best that the BBP guidelines are checked.

Reply:
The guidelines are here: https://bbpteam.epfl.ch/project/spaces/display/BBPSTD/7-2.+OSS+Quality+Requirements#id-72.OSSQualityRequirements-C-LicenseNotice

Reply:
that's true! I was looking at existing open-sourced software and I didn't see that has been followed consistently (at least a few I checked).

But I see that we have internally documented it in https://bbpteam.epfl.ch/project/spaces/display/BBPSTD/7-2.+OSS+Quality+Requirements#id-72.OSSQualityRequirements-C-LicenseNotice!

So I think we can go ahead with similar to Spack:

```console
/*
 * Copyright 2023 Blue Brain Project, EPFL.
 * See the top-level LICENSE file for details.
 *
 *  SPDX-License-Identifier: Apache-2.0
 */
```